doc/specs/dna.md

```diff
@@ -0,0 +1,160 @@
+// Note that you write validation by checking the type in the call to
+// `validateLink`
+commit("post",{content:"my post"})
+commit("post_link",[
+    {base:post,tag:"handle",target:myHandle},
+    {base:myHandle,tag:"posts",target:post}
+    ])
+commit("tag_link",[
+    {base:myHandle,tag:"_tag"+theTag, target:post}
+    {base:post,tag:"taggedBy",target:myHandle},
+])
+
+// New Way:
+// NOte: that validation code is in the entry type `links-to` block
+// which gets found by lookup via the type of the base and the tag.
+let bundle = startBundle()
+let post = commit("post",{content:"my post"})
+link(post, "author", myHandle)
+link(myHandle, "posts", post)
+link(myHandle, "_tag"+theTag, post)
+link(post,"taggedBy", myHandle)
+closeBundle(bundle)
+```
+
+## Example DNA Tree:
+
+``` javascript
+{
+    "name": "Example app",
+    "version": "0.0.1",
+    "uuid": "123e4567-e89b-12d3-a456-426655440000",
+    "dna_spec_version": "2.0",
+    "dht_config": {
+    },
+    "properties": {
+        "weight": "2kg"
+    },
+    "zomes": [
+        {
+            "name": "clutter",
+            "description": "zome that implements micro-blogging",
+            "config": {
+                "error_handling": "throw-errors"
+            },
+            //"RibosomeType":"WASM", #do we just commit to WASM only at
+            //  at this stage?
+            "entry-types": [
+
+                {
+                    "name": "post",
+                    "schema": {}, // maybe NOT because happens inside WASM validations and it's fast enought.
+
+                    "validation": ".." // WASM code
+
+                    //"semantics":{}, #??? UI hints do this later
+                    "sharing": "public", ///private/encrypted
+                    "links_to": [
+                        {
+                            "target_type": "handle",
+                            "tag": "author",
+                            "validation": ".." //WASM
+                        }
+                    ]
+                },
+                {
+                    "name": "handle",
+                    "links_to": [
+                        {
+                            "target_type": "post",
+                            "tag": "posts",
+                            "validation": ".." //WASM
+                        },
+                        {
+                            "target_type": "post",
+                            "tag": "_tag*",
+                            "validation": ".." //WASM
+                        }
+
+                        ...
+                    ]
+                },
+                {
+                    "name": "handle_link",
+                    "schema": {
+                        "role": "string"
+                    }
+                }
+            ], // end of entry-types
+            "capabilities": [
+                {
+                    "name": "web_gateway",
+                    "capability": {
+                        "membrane": "public"
```

Comment:
I'd rename "membrane" to "domain" because I see membrane as a gating process like for instance a validation function.